src/graph.js

```diff
@@ -40,37 +40,112 @@ class Graph {
   // Optionally accepts an array of other GraphNodes for the new vertex to be connected to
   // Returns the newly-added vertex
   addVertex(value, edges = []) {
-
+    const newNode = new GraphNode({ value, edges });
+    edges.forEach((edge) => {
+      edge.pushToEdges(newNode);
+    });
+    this.vertices.push(newNode);
+    if (this.vertices.length === 2) {
+      this.vertices[0].pushToEdges(this.vertices[1]);
+      this.vertices[1].pushToEdges(this.vertices[0]);
+    }
+    return newNode;
   }
   // Checks all the vertices of the graph for the target value
   // Returns true or false
   contains(value) {
-
+    let doesContain = false;
+    this.vertices.forEach((vert) => {
+      if (vert.value === value) {
+        doesContain = true;
+      }
+    });
+    return doesContain;
   }
   // Checks the graph to see if a GraphNode with the specified value exists in the graph 
   // and removes the vertex if it is found
   removeVertex(value) {
-
+    for (let i = 0; i < this.vertices.length; i++) {
+      if (this.vertices[i].value === value) {
+        this.vertices.splice(i, 1);
+      }
+    }
   }
   // Checks the two input vertices to see if each one references the other in their respective edges array
   // Both vertices must reference each other for the edge to be considered valid
   // If only one vertex references the other but not vice versa, should not return true
   // Note: You'll need to store references to each vertex's array of edges so that you can use 
   // array methods on said arrays. There is no method to traverse the edge arrays built into the GraphNode class
   checkIfEdgeExists(fromVertex, toVertex) {
-
+    const fromEdges = fromVertex.edges;
+    const toEdges = toVertex.edges;
+    let match1 = false; 
+    let match2 = false;
+    fromEdges.forEach((from) => {
+      toEdges.forEach((to) => {
+        if (fromVertex === to) {
+          match1 = true;
+        }
+        if (toVertex === from) {
+          match2 = true;
+        }
+      });
+    });
+    if (match1 && match2) {
+      return true;
+    }
+    return false;
   }
   // Adds an edge between the two given vertices if no edge already exists between them
   // Again, an edge means both vertices reference the other 
   addEdge(fromVertex, toVertex) {
-
+    if (!this.checkIfEdgeExists(fromVertex, toVertex)) {
+      if (!toVertex.edges.includes(fromVertex)) {
+        toVertex.pushToEdges(fromVertex);
+      }
+      if (!fromVertex.edges.includes(toVertex)) {
+        fromVertex.pushToEdges(toVertex);
+      }
+    }
   }
   // Removes the edge between the two given vertices if an edge already exists between them
   // After removing the edge, neither vertex should be referencing the other
   removeEdge(fromVertex, toVertex) {
-
+    const fromEdges = fromVertex.edges;
+    const toEdges = toVertex.edges;
+    console.log(fromEdges);
```

Comment:
Make sure you remove console logs when committing code and making pull requests.